lib/middleware/checkLicenceStatus.js

```diff
@@ -1,58 +1,66 @@
 const logger = require('../logger');
-const accessLicenceService = require('../services/accessLicenceService');
-const acquisitionContextService = require('../services/acquisitionContextService');
-
-function* _updateLicenceList(req, res) {
-  let licenceList = yield accessLicenceService.getLicenceList({adminuserid:req.currentUser.uuid});
-  logger.info({operation:'updateLicenceList', licenceId:req.licenceId, isLicenceListArray: licenceList.length});
-
-  if (licenceList && licenceList.length >= 0) {
-    logger.info({operation:'updateLicenceList', licenceId:req.licenceId, isLicenceListArray: licenceList.length});
-    //if licenceList exists and has properties return the array without the current licenceId
-      licenceList = licenceList.filter((item) => {
-        if (item.licenceId !== req.licenceId) {
-        return item;
-        }
-    });
-    //decorate the request with the next licenceId in the list
-    req.licenceId = licenceList[0].licenceId;
-		logger.info({operation:'updatedLicenceList', newLicenceSelected:req.licenceId, newLicenceListArray: licenceList.length});
-  } else {
-    const err = new Error('Licence has been revoked');
-    err.status = 403;
-    throw err;
-  }
-}
+const proxies = require('kat-client-proxies');
+const accessLicenceClient = proxies.accessLicenceClient;
+const acquisitionCtxClient = proxies.acquisitionCtxClient;
+const getListOfLicences = require('./getListOfLicences');
+
+function _goToNextLicence(req, res) {
+    const operation = 'goToNextLicence';
+    logger.info({operation, licenceId: req.licenceId});
 
-function* isActive(req, res, next) {
-    try {
+    return getListOfLicences(req)
+        .then(() => {
+            logger.info({operation, licenceId: req.licenceId, isLicenceListArray: req.listOfLicences.length});
 
-        const currentLicence = yield accessLicenceService.getLicenceInfo(req.licenceId);
-        logger.info({operation:'checkLicenceStatus', licenceStatus: currentLicence.status});
-        if (currentLicence.status !== 'active') {
-          //Check if currentLicence is in cached listOfLicences
-          _updateLicenceList(req, res);
-          }
+            let newLicence = undefined;
+            req.listOfLicences.every(item => {
```

Comment:
The reason why I'm using `.every()` here is because we don't need to go through all the array elements. Once our condition fulfills, we can stop parsing and `.forEach()` can't do that (`.every()` can :) )